R/bayesiannetworkanalysis.R

```diff
@@ -0,0 +1,1297 @@
+    for (i in seq_len(nGraphs)) {
+      
+      # Check with values to add to the edge evidence table: 
+      if (options$evidenceType == "inclusionProbability") {
+        toAdd <- allNetworks[[i]][["inclusionProbabilities"]]
+      } else if (options$evidenceType == "BF10") {
+        toAdd <- allNetworks[[i]][["BF"]]
+      } else if (options$evidenceType == "BF01") {
+        toAdd <- 1 / allNetworks[[i]][["BF"]]
+      } else {
+        toAdd <- log(allNetworks[[i]][["BF"]])
+      }
+      
+      toAdd <- as.data.frame(toAdd)
+      names(toAdd) <- paste0(variables, i)
+      
+      TBcolumns <- cbind(TBcolumns, toAdd)
+    }
+    table$setData(TBcolumns)
+  }
+  mainContainer[["edgeEvidenceTable"]] <- table
+}
+
+.bayesianNetworkAnalysisOneStructurePlot <- function(network, options, layout, 
+                                                     groups, labels, legend, shape, nodeColor, nodeNames) {
+  
+  return(
+    qgraph::qgraph(
+      input               = network[["structure"]],
+      layout              = layout,
+      groups              = groups,
+      repulsion           = options[["repulsion"]],
+      cut                 = options[["cut"]],
+      edge.width          = options[["edgeSize"]],
+      node.width          = options[["nodeSize"]],
+      details             = options[["showDetails"]],
+      labels              = labels,
+      palette             = if (options[["manualColors"]]) NULL else options[["nodePalette"]],
+      legend              = legend,
+      shape               = shape,
+      color               = nodeColor,
+      nodeNames           = nodeNames,
+      label.scale         = options[["scaleLabels"]],
+      label.cex           = options[["labelSize"]],
+      GLratio             = 1 / options[["legendToPlotRatio"]],
+      edge.labels         = options[["edgeLabels"]],
+      edge.label.cex      = options[["edgeLabelCex"]],
+      edge.label.position = options[["edgeLabelPosition"]]
+    ))
+}
+
+# =========================
+#  ADDITIONAL FUNCTIONS
+# =========================
+
+# Turns vector into matrix
+vectorToMatrix <- function(vec, p, diag = FALSE, bycolumn = FALSE) {
+  m <- matrix(0, p, p)
+  
+  if(bycolumn == F){
+    m[lower.tri(m, diag = diag)] <- vec
+    m <- t(m)
+    m[lower.tri(m)] <- t(m)[lower.tri(m)]
+  } else {
+    m[upper.tri(m, diag = diag)] <- vec
+    m <- t(m)
+    m[upper.tri(m)] <- t(m)[upper.tri(m)]
+  }
+  return(m)
+}
+
+# Transform precision into partial correlations for interpretation
+pr2pc <- function(K) {
+  D.Prec = diag(diag(K)^(-.5))
+  R <- diag(2,dim(K)[1])-D.Prec%*%K%*%D.Prec
+  colnames(R) <- colnames(K)
+  rownames(R) <- rownames(K)
+  return(R)
+}
+
+# BDgraph stores graphs as byte strings for efficiency
+string2graph <- function(Gchar, p) {
+  Gvec = rep(0, p*(p-1)/2)
+  edges <- which(unlist(strsplit(as.character(Gchar), "")) == 1)
+  Gvec[edges] = 1
+  G <- matrix(0, p, p)
+  G[upper.tri(G)] <- Gvec
+  G = G + t(G)
```

Comment:
```suggestion
  Gvec <- rep(0, p*(p-1)/2)
  edges <- which(unlist(strsplit(as.character(Gchar), "", fixed = TRUE)) == 1)
  Gvec[edges] = 1
  G <- matrix(0, p, p)
  G[upper.tri(G)] <- Gvec
  G <- G + t(G)
```
if `fixed = FALSE` (the default) then `""` is interpreted as a regular expression. Also `<-` instead of `=`